cmd/attest.go

```diff
@@ -0,0 +1,218 @@
+key or AK for a custom attestation key.By default it uses gceAK.
+--algo flag override the public key algorithm for attestation key. If not provided then
+by default rsa is used.
+--nonce attaches even length extra data to the report and guarantees a fresh quote.
+--teenonce attaches a 64 bytes extra data to the attestation report of hardware and 
+guarantees a fresh quote.
+--output flag provides file name and location to store attestation report in the binary
+format. If filename is not provided then output will be printed on stdout.
+--outputText flag can be used to add text format of attestation report in a file.
+--gce-project-id and --gce-project-number used to verify attestation report.Either both
+will be used or neither.
+--gce-zone , --gce-instance-id and --gce-instance-name are optional flags.
+`,
+	Args: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		rwc,err := openTpm()
+		if(err != nil){
+			return err
+		}
+		defer rwc.Close()
+
+
+		createFunc := attestationKeys[key][keyAlgo]
+		
+		attestationKey , err := createFunc(rwc)
+		if err != nil {
+			return fmt.Errorf("failed to create attestation key: %v", err)
+		}
+		defer attestationKey.Close()
+
+
+		attestopts := client.AttestOpts{}
+		attestopts.Nonce = nonce
+		if(len(teenonce) != 0){
+			attestopts.TEENonce = teenonce
+			attestopts.TEEDevice,err = client.CreateSevSnpDevice()
+			if(err != nil){
+				return fmt.Errorf("collecting TEE attestation report: %v",err)
+			}
+		}		
+
+
+		attestation ,err := attestationKey.Attest(attestopts)
+		if(err != nil){
+			return fmt.Errorf("while collecting attestation report : %v",err)
+		}
+
+
+		if key == "gceAK"{
+			if(len(gceProjectID) != 0 && gceProjectNumber != 0){
+				getInstanceMetadata()
+				addGceInstanceInfo(attestation)
+			}
+			if((len(gceProjectID) == 0 && gceProjectNumber != 0) || (len(gceProjectID) != 0 && gceProjectNumber == 0)){
+				return fmt.Errorf("attestation key type gceAK either use gceProjectId and gceProjectNumber together or none of them")
+			}
+
+		}else{
+			if(len(gceProjectID) != 0 || gceProjectNumber != 0){
+				return fmt.Errorf("attestation key type AK cannot use Gce Flags")
+			}
+		}
+		
+
+		if(len(outputText) != 0){
+			protoOut := []byte(prototext.Format(attestation))
+			if _,err := outputProto().Write(protoOut); err != nil{
+				return fmt.Errorf("failed to write attesattion report in textProto format: %v",err)
+			}
+		}
+
+		
+		if(len(output) != 0){
+			out ,err:= proto.Marshal(attestation)
+			if err != nil {
+				return fmt.Errorf("failed to marshal attestation proto: %v", attestation)
+			}
+			if _, err := dataOutput().Write(out); err != nil {
+					return fmt.Errorf("failed to write attestation proto: %v", err)
+			}
+		}
+		return nil
+	},
+	
+}
+
+func getInstanceMetadata() error{
+	if(len(gceZone) == 0){
+		out ,err := exec.Command("curl", "http://metadata.google.internal/computeMetadata/v1/instance/zone").Output()
```

Comment:
I have used some string slicing to get the correct value for Zone.
But as you suggested , we can get this using https://pkg.go.dev/cloud.google.com/go/compute/metadata 
package.
So  I'll change the implementation and use [Zone()](https://pkg.go.dev/cloud.google.com/go/compute/metadata#Client.Zone) instead of using curl command.